test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -721,10 +726,15 @@
                     certname]
                    {:as-arrays? true}))))))))
 
-  (deftest facts-blacklist
+  (deftest regex-facts-blacklist
     (dotestseq [version fact-versions
-                :let [command (update facts :values #(assoc % "blacklisted-fact" "val"))]]
-      (testing "should ignore the blacklisted fact"
+                :let [command (update facts :values
+                                      #(assoc % "blacklisted-fact" "val"
+                                              "prefix" "val"
+                                              "facts-suff" "val"
+                                              "glob" "val"
+                                              "puppet" "val"))]]
```

Comment:
Since it's cheap, might be worth adding a "suff-nope" and maybe a "*blacklisted-fact*" just those other two cases we expect to pass through (i.e. trailing, and "all").